src/components/Problem/ProblemCard.vue

```diff
@@ -55,28 +60,37 @@ const session = useSession();
                 </div>
               </div>
             </div>
-            <template v-if="!preview">
+
+            <div class="ml-3 flex flex-wrap place-items-center gap-x-3" v-if="!preview">
               <router-link
-                class="btn ml-3 md:btn-md lg:btn-lg"
+                class="btn md:btn-md lg:btn-lg"
                 :to="`/course/${$route.params.name}/problem/${$route.params.id}/submit`"
               >
-                <i-uil-file-upload-alt class="mr-1 lg:h-5 lg:w-5" /> Submit
+                <i-uil-file-upload-alt class="lg:h-5 lg:w-5" /> Submit
               </router-link>
               <router-link
-                class="btn ml-3 md:btn-md lg:btn-lg"
+                class="btn md:btn-md lg:btn-lg"
                 :to="`/course/${$route.params.name}/problem/${$route.params.id}/stats`"
               >
-                <i-uil-chart-line class="mr-1 lg:h-5 lg:w-5" /> Stats
+                <i-uil-chart-line class="lg:h-5 lg:w-5" /> Stats
               </router-link>
               <router-link
                 v-if="session.isAdmin"
-                :class="['btn-ghost tooltip tooltip-bottom btn-sm btn ml-3', 'inline-flex']"
+                :class="['btn-ghost tooltip tooltip-bottom btn-sm btn', 'inline-flex']"
                 data-tip="Copycat"
                 :to="`/course/${$route.params.name}/problem/${$route.params.id}/copycat`"
               >
-                <i-uil-file-exclamation-alt class="mr-1 lg:h-5 lg:w-5" />
+                <i-uil-file-exclamation-alt class="lg:h-5 lg:w-5" />
               </router-link>
-            </template>
+              <button
+                v-if="session.isAdmin"
+                :class="['btn-ghost tooltip tooltip-bottom btn-sm btn', 'inline-flex']"
+                data-tip="Download test case"
+                @click="downloadTestCase(Number.parseInt($route.params.id as string))"
```

Comment:
I'm not sure that if passing `radix` in `Number.parseInt` is STILL recommended or not, but let's pass it as a convention.

Check this out
https://developer.mozilla.org/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/parseInt

Reply:
done 👍